packages/core/index.ts

```diff
@@ -11,18 +11,35 @@ type Impossible<T extends Record<string, any>> = Partial<
   Record<keyof T, never>
 >;
 
-export interface BaseOptions<TShared extends Record<string, ZodType>> {
+export interface BaseOptions<
+  TShared extends Record<string, ZodType>,
+  TExtends extends Record<string, unknown> | undefined
```

Comment:
Some suggestion, we can use TS "Branded Types" pattern to prevent user from passing non ts-env `env` into `extends`, if we really want to restrict that🙂

Reply:
I'm open to this, but do you think it's really necessary?

Reply:
As far as I can tell from your change I don't think it's necessary, just need to keep in might is that this pretty much allow user to pass in random object to the `extends`.